SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -4311,63 +4309,59 @@ void CEulerSolver::SetFarfield_AoA(CGeometry *geometry, CSolver **solver_contain
 
     /*--- Compute a new value for AoA on the fine mesh only (degrees)---*/
 
-    if (iMesh == MESH_0) AoA = AoA + AoA_inc;
-    else { AoA = config->GetAoA(); }
-
-    /*--- Only the fine mesh stores the updated values for AoA in config ---*/
-
     if (iMesh == MESH_0) {
+      AoA = AoA + AoA_inc;
       config->SetAoA(AoA);
     }
+    else {
+      AoA = config->GetAoA();
+    }
+
+    AoA *= PI_NUMBER/180.0;
 
     /*--- Update the freestream velocity vector at the farfield ---*/
 
-    for (iDim = 0; iDim < nDim; iDim++)
-      Vel_Infty[iDim] = GetVelocity_Inf(iDim);
+    su2double Vel_Infty[MAXNDIM]={0.0};
 
-    /*--- Compute the magnitude of the free stream velocity ---*/
+    for (auto iDim = 0u; iDim < nDim; iDim++)
+      Vel_Infty[iDim] = GetVelocity_Inf(iDim);
 
-    Vel_Infty_Mag = 0;
-    for (iDim = 0; iDim < nDim; iDim++)
-      Vel_Infty_Mag += Vel_Infty[iDim]*Vel_Infty[iDim];
-    Vel_Infty_Mag = sqrt(Vel_Infty_Mag);
+    const su2double Vel_Infty_Mag = GeometryToolbox::Norm(int(MAXNDIM), Vel_Infty);
 
     /*--- Compute the new freestream velocity with the updated AoA ---*/
 
     if (nDim == 2) {
-      Vel_Infty[0] = cos(AoA*PI_NUMBER/180.0)*Vel_Infty_Mag;
-      Vel_Infty[1] = sin(AoA*PI_NUMBER/180.0)*Vel_Infty_Mag;
+      Vel_Infty[0] = cos(AoA)*Vel_Infty_Mag;
+      Vel_Infty[1] = sin(AoA)*Vel_Infty_Mag;
     }
-    if (nDim == 3) {
-      Vel_Infty[0] = cos(AoA*PI_NUMBER/180.0)*cos(Beta*PI_NUMBER/180.0)*Vel_Infty_Mag;
-      Vel_Infty[1] = sin(Beta)*Vel_Infty_Mag;
-      Vel_Infty[2] = sin(AoA*PI_NUMBER/180.0)*cos(Beta*PI_NUMBER/180.0)*Vel_Infty_Mag;
+    else {
+      Vel_Infty[0] = cos(AoA)*cos(AoS)*Vel_Infty_Mag;
+      Vel_Infty[1] = sin(AoS)*Vel_Infty_Mag;
```

Comment:
Service comment: This line is the bugfix of discussion #1298 and subsequent issue #1301, the remainder of this commit is just nice clean-up 🧹 (please correct me if i am wrong)